firmware-support/bittide-sys/src/dna_port_e2.rs

```diff
@@ -0,0 +1,33 @@
+// SPDX-FileCopyrightText: 2024 Google LLC
+//
+// SPDX-License-Identifier: Apache-2.0
+/*! DNA Port E2 peripheral interface.
+
+This module contains an interface to read the DNA port E2 peripheral.
+The DNA value is a 96-bit unique identifier per device, the interface to device
+identifier is offered by the `DnaPortE2` struct.
+*/
+
+/// The DnaPortE2 struct is a hardware abstraction for the DnaPortE2 peripheral,
+/// a unqique identifier for the FPGA device.
+pub struct DnaPortE2 {}
+
+impl DnaPortE2 {
+    /// Reads the DNA value from the DNA port E2 peripheral.
+    /// The DNA value is a 96-bit unique identifier, the upper bits from the 128-bit value are zero.
+    pub fn read(addr: usize) -> u128 {
```

Comment:
Could this be a freestanding function? Does this need the struct?

Reply:
We could indeed just make a freestanding function.